### ISSUE TITLE
Missing software for this to run? Worldserver and authserver autoclose.

### ISSUE THREAD

**Issue by Kuiriel** [ZhengPeiRu21/mod-individual-progression#71]
TL;DR: Something about default set up of the released preview build does not function due to inability to connect to MySQL database?

\
Hi. I was able to get a WoW server running via ChromieCraft's basic Docker Desktop guide. I was confused as there was no obvious CONF settings exposed for me to tweak and I didn't know how to get started with CMake etc, and I figured I would leave that lesson for once I have a clue.

So I thought I would try your Module Preview Build at https://github.com/ZhengPeiRu21/mod-individual-progression/releases out - it openly exposes more settings and the mods come pre-installed in. For example, I was hoping old world elites might be restored for low level tears.

However, I haven't been able to get the world server up and running. World server and Auth Server both close instantly when I run them or after executing play.bat.

I'm pretty sure I followed the install instructions correctly. I had Docker Desktop service disabled and not running at the time, but I launched it, went in and made sure to docker-compose stop as well. Do I need to have it removed entirely?

I have tried running it from an admin command prompt as well, which removed the mysqld error but did not solve the rest.

What am I missing? Using default worldserver and authserver conf settings:

```
mysqld: Could not create or access the registry key needed for the MySQL application
to log to the Windows EventLog. Run the application with sufficient
privileges once to create the key, add the key manually, or turn off
logging for that application.
2022-12-29T06:40:08.602990Z 0 [Warning] TIMESTAMP with implicit DEFAULT value is deprecated. Please use --explicit_defaults_for_timestamp server option (see documentation for more details).
2022-12-29T06:40:08.603034Z 0 [Warning] 'NO_ZERO_DATE', 'NO_ZERO_IN_DATE' and 'ERROR_FOR_DIVISION_BY_ZERO' sql modes should be used with strict mode. They will be merged with strict mode in a future release.
2022-12-29T06:40:08.604025Z 0 [ERROR] Cannot open Windows EventLog; check privileges, or start server with --log_syslog=0
2022-12-29T06:40:08.604037Z 0 [Note] .\Database\MySQL Server 5.7\bin\mysqld.exe (mysqld 5.7.30-log) starting as process 1964 ...
2022-12-29T06:40:08.606117Z 0 [Note] InnoDB: Mutexes and rw_locks use Windows interlocked functions
2022-12-29T06:40:08.606255Z 0 [Note] InnoDB: Uses event mutexes
2022-12-29T06:40:08.606296Z 0 [Note] InnoDB: _mm_lfence() and _mm_sfence() are used for memory barrier
2022-12-29T06:40:08.606330Z 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
2022-12-29T06:40:08.606369Z 0 [Note] InnoDB: Adjusting innodb_buffer_pool_instances from 8 to 1 since innodb_buffer_pool_size is less than 1024 MiB
2022-12-29T06:40:08.606544Z 0 [Note] InnoDB: Number of pools: 1
2022-12-29T06:40:08.606662Z 0 [Note] InnoDB: Not using CPU crc32 instructions
2022-12-29T06:40:08.607688Z 0 [Note] InnoDB: Initializing buffer pool, total size = 8M, instances = 1, chunk size = 8M
2022-12-29T06:40:08.607974Z 0 [Note] InnoDB: Completed initialization of buffer pool
2022-12-29T06:40:08.623720Z 0 [Note] InnoDB: Highest supported file format is Barracuda.
2022-12-29T06:40:08.687273Z 0 [Note] InnoDB: Creating shared tablespace for temporary tables
2022-12-29T06:40:08.687475Z 0 [Note] InnoDB: Setting file '.\ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
2022-12-29T06:40:08.690410Z 0 [Note] InnoDB: File '.\ibtmp1' size is now 12 MB.
2022-12-29T06:40:08.693663Z 0 [Note] InnoDB: 96 redo rollback segment(s) found. 96 redo rollback segment(s) are active.
2022-12-29T06:40:08.693715Z 0 [Note] InnoDB: 32 non-redo rollback segment(s) are active.
2022-12-29T06:40:08.693899Z 0 [Note] InnoDB: Waiting for purge to start
2022-12-29T06:40:08.746679Z 0 [Note] InnoDB: 5.7.30 started; log sequence number 10591431714
2022-12-29T06:40:08.747017Z 0 [Note] Plugin 'FEDERATED' is disabled.
2022-12-29T06:40:08.747086Z 0 [Note] InnoDB: Loading buffer pool(s) from C:\Games\WoW\mod-individual-progression\Database\Data\ib_buffer_pool
2022-12-29T06:40:08.751252Z 0 [Note] Found ca.pem, server-cert.pem and server-key.pem in data directory. Trying to enable SSL support using them.
2022-12-29T06:40:08.751309Z 0 [Note] Skipping generation of SSL certificates as certificate files are present in data directory.
2022-12-29T06:40:08.752033Z 0 [Warning] CA certificate ca.pem is self signed.
2022-12-29T06:40:08.752155Z 0 [Note] Skipping generation of RSA key pair as key files are present in data directory.
2022-12-29T06:40:08.752295Z 0 [Note] Server hostname (bind-address): '*'; port: 3306
2022-12-29T06:40:08.752443Z 0 [Note] IPv6 is available.
2022-12-29T06:40:08.752490Z 0 [Note]   - '::' resolves to '::';
2022-12-29T06:40:08.752632Z 0 [Note] Server socket created on IP: '::'.
2022-12-29T06:40:08.753825Z 0 [Note] InnoDB: Buffer pool(s) load completed at 221229 17:40:08
2022-12-29T06:40:08.759182Z 0 [Note] Failed to start slave threads for channel ''
2022-12-29T06:40:08.764227Z 0 [Note] Event Scheduler: Loaded 0 events
2022-12-29T06:40:08.764360Z 0 [Note] .\Database\MySQL Server 5.7\bin\mysqld.exe: ready for connections.
Version: '5.7.30-log'  socket: ''  port: 3306  MySQL Community Server (GPL)
2022-12-29T06:40:11.588138Z 3 [Note] Access denied for user 'acore'@'localhost' (using password: YES)
```

Using my server settings where I provide my local IP address so that clients will be able to connect, the errors are slightly different.

```
mysqld: Could not create or access the registry key needed for the MySQL application
to log to the Windows EventLog. Run the application with sufficient
privileges once to create the key, add the key manually, or turn off
logging for that application.
2022-12-29T06:38:08.884660Z 0 [Warning] TIMESTAMP with implicit DEFAULT value is deprecated. Please use --explicit_defaults_for_timestamp server option (see documentation for more details).
2022-12-29T06:38:08.884697Z 0 [Warning] 'NO_ZERO_DATE', 'NO_ZERO_IN_DATE' and 'ERROR_FOR_DIVISION_BY_ZERO' sql modes should be used with strict mode. They will be merged with strict mode in a future release.
2022-12-29T06:38:08.885782Z 0 [ERROR] Cannot open Windows EventLog; check privileges, or start server with --log_syslog=0
2022-12-29T06:38:08.885796Z 0 [Note] .\Database\MySQL Server 5.7\bin\mysqld.exe (mysqld 5.7.30-log) starting as process 22660 ...
2022-12-29T06:38:08.887810Z 0 [Note] InnoDB: Mutexes and rw_locks use Windows interlocked functions
2022-12-29T06:38:08.887846Z 0 [Note] InnoDB: Uses event mutexes
2022-12-29T06:38:08.887873Z 0 [Note] InnoDB: _mm_lfence() and _mm_sfence() are used for memory barrier
2022-12-29T06:38:08.887901Z 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
2022-12-29T06:38:08.887935Z 0 [Note] InnoDB: Adjusting innodb_buffer_pool_instances from 8 to 1 since innodb_buffer_pool_size is less than 1024 MiB
2022-12-29T06:38:08.888115Z 0 [Note] InnoDB: Number of pools: 1
2022-12-29T06:38:08.888205Z 0 [Note] InnoDB: Not using CPU crc32 instructions
2022-12-29T06:38:08.889188Z 0 [Note] InnoDB: Initializing buffer pool, total size = 8M, instances = 1, chunk size = 8M
2022-12-29T06:38:08.889481Z 0 [Note] InnoDB: Completed initialization of buffer pool
2022-12-29T06:38:08.903414Z 0 [Note] InnoDB: Highest supported file format is Barracuda.
2022-12-29T06:38:08.968461Z 0 [Note] InnoDB: Creating shared tablespace for temporary tables
2022-12-29T06:38:08.968841Z 0 [Note] InnoDB: Setting file '.\ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
2022-12-29T06:38:08.971829Z 0 [Note] InnoDB: File '.\ibtmp1' size is now 12 MB.
2022-12-29T06:38:08.974993Z 0 [Note] InnoDB: 96 redo rollback segment(s) found. 96 redo rollback segment(s) are active.
2022-12-29T06:38:08.975041Z 0 [Note] InnoDB: 32 non-redo rollback segment(s) are active.
2022-12-29T06:38:08.975208Z 0 [Note] InnoDB: Waiting for purge to start
2022-12-29T06:38:09.026683Z 0 [Note] InnoDB: 5.7.30 started; log sequence number 10591431670
2022-12-29T06:38:09.027123Z 0 [Note] InnoDB: Loading buffer pool(s) from C:\Games\WoW\mod-individual-progression\Database\Data\ib_buffer_pool
2022-12-29T06:38:09.027158Z 0 [Note] Plugin 'FEDERATED' is disabled.
2022-12-29T06:38:09.032480Z 0 [Note] Found ca.pem, server-cert.pem and server-key.pem in data directory. Trying to enable SSL support using them.
2022-12-29T06:38:09.032540Z 0 [Note] Skipping generation of SSL certificates as certificate files are present in data directory.
2022-12-29T06:38:09.033266Z 0 [Warning] CA certificate ca.pem is self signed.
2022-12-29T06:38:09.033369Z 0 [Note] Skipping generation of RSA key pair as key files are present in data directory.
2022-12-29T06:38:09.033523Z 0 [Note] Server hostname (bind-address): '*'; port: 3306
2022-12-29T06:38:09.033659Z 0 [Note] IPv6 is available.
2022-12-29T06:38:09.033703Z 0 [Note]   - '::' resolves to '::';
2022-12-29T06:38:09.033770Z 0 [Note] Server socket created on IP: '::'.
2022-12-29T06:38:09.034687Z 0 [Note] InnoDB: Buffer pool(s) load completed at 221229 17:38:09
2022-12-29T06:38:09.040262Z 0 [Note] Failed to start slave threads for channel ''
2022-12-29T06:38:09.045036Z 0 [Note] Event Scheduler: Loaded 0 events
2022-12-29T06:38:09.045152Z 0 [Note] .\Database\MySQL Server 5.7\bin\mysqld.exe: ready for connections.
Version: '5.7.30-log'  socket: ''  port: 3306  MySQL Community Server (GPL)
```

Using default settings with an admin command prompt as well, which removed the mysqld error but did not solve the rest.

```
2022-12-29T06:57:27.399414Z 0 [Warning] TIMESTAMP with implicit DEFAULT value is deprecated. Please use --explicit_defaults_for_timestamp server option (see documentation for more details).
2022-12-29T06:57:27.399450Z 0 [Warning] 'NO_ZERO_DATE', 'NO_ZERO_IN_DATE' and 'ERROR_FOR_DIVISION_BY_ZERO' sql modes should be used with strict mode. They will be merged with strict mode in a future release.
2022-12-29T06:57:27.400497Z 0 [Note] .\Database\MySQL Server 5.7\bin\mysqld.exe (mysqld 5.7.30-log) starting as process 3228 ...
2022-12-29T06:57:27.402684Z 0 [Note] InnoDB: Mutexes and rw_locks use Windows interlocked functions
2022-12-29T06:57:27.402807Z 0 [Note] InnoDB: Uses event mutexes
2022-12-29T06:57:27.402874Z 0 [Note] InnoDB: _mm_lfence() and _mm_sfence() are used for memory barrier
2022-12-29T06:57:27.402935Z 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
2022-12-29T06:57:27.403009Z 0 [Note] InnoDB: Adjusting innodb_buffer_pool_instances from 8 to 1 since innodb_buffer_pool_size is less than 1024 MiB
2022-12-29T06:57:27.403304Z 0 [Note] InnoDB: Number of pools: 1
2022-12-29T06:57:27.403477Z 0 [Note] InnoDB: Not using CPU crc32 instructions
2022-12-29T06:57:27.404640Z 0 [Note] InnoDB: Initializing buffer pool, total size = 8M, instances = 1, chunk size = 8M
2022-12-29T06:57:27.405020Z 0 [Note] InnoDB: Completed initialization of buffer pool
2022-12-29T06:57:27.420047Z 0 [Note] InnoDB: Highest supported file format is Barracuda.
2022-12-29T06:57:27.485783Z 0 [Note] InnoDB: Creating shared tablespace for temporary tables
2022-12-29T06:57:27.486274Z 0 [Note] InnoDB: Setting file '.\ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
2022-12-29T06:57:27.489263Z 0 [Note] InnoDB: File '.\ibtmp1' size is now 12 MB.
2022-12-29T06:57:27.492942Z 0 [Note] InnoDB: 96 redo rollback segment(s) found. 96 redo rollback segment(s) are active.
2022-12-29T06:57:27.493116Z 0 [Note] InnoDB: 32 non-redo rollback segment(s) are active.
2022-12-29T06:57:27.493344Z 0 [Note] InnoDB: Waiting for purge to start
2022-12-29T06:57:27.556851Z 0 [Note] InnoDB: 5.7.30 started; log sequence number 10591431798
2022-12-29T06:57:27.557398Z 0 [Note] InnoDB: Loading buffer pool(s) from C:\Games\WoW\mod-individual-progression\Database\Data\ib_buffer_pool
2022-12-29T06:57:27.557443Z 0 [Note] Plugin 'FEDERATED' is disabled.
2022-12-29T06:57:27.563152Z 0 [Note] Found ca.pem, server-cert.pem and server-key.pem in data directory. Trying to enable SSL support using them.
2022-12-29T06:57:27.563292Z 0 [Note] Skipping generation of SSL certificates as certificate files are present in data directory.
2022-12-29T06:57:27.564088Z 0 [Warning] CA certificate ca.pem is self signed.
2022-12-29T06:57:27.564253Z 0 [Note] Skipping generation of RSA key pair as key files are present in data directory.
2022-12-29T06:57:27.564475Z 0 [Note] Server hostname (bind-address): '*'; port: 3306
2022-12-29T06:57:27.564661Z 0 [Note] IPv6 is available.
2022-12-29T06:57:27.564745Z 0 [Note]   - '::' resolves to '::';
2022-12-29T06:57:27.564868Z 0 [Note] Server socket created on IP: '::'.
2022-12-29T06:57:27.565387Z 0 [Note] InnoDB: Buffer pool(s) load completed at 221229 17:57:27
2022-12-29T06:57:27.571775Z 0 [Note] Failed to start slave threads for channel ''
2022-12-29T06:57:27.577096Z 0 [Note] Event Scheduler: Loaded 0 events
2022-12-29T06:57:27.577308Z 0 [Note] .\Database\MySQL Server 5.7\bin\mysqld.exe: ready for connections.
Version: '5.7.30-log'  socket: ''  port: 3306  MySQL Community Server (GPL)
2022-12-29T06:57:30.346620Z 2 [Note] Access denied for user 'acore'@'localhost' (using password: YES)
2022-12-29T06:57:30.503621Z 3 [Note] Access denied for user 'acore'@'localhost' (using password: YES)
```

When I changed the login details from root;; to acore;acore; like in the listed defaults, I no longer saw the access denied error. Both authserver and worldserver opened up for longer but still vanished. So I loaded them directly in command prompt and this was the error with the defaults:

```
Could not connect to MySQL database at 127.0.0.1: Access denied for user 'acore'@'localhost' (using password: YES)
DatabasePool Login NOT opened. There were errors opening the MySQL connections. Check your log file for specific errors
```

and a slightly different error with my custom IP settings. 

```
Could not connect to MySQL database at 192.168.0.118: Can't connect to MySQL server on '192.168.0.118' (10061)
DatabasePool Login NOT opened. There were errors opening the MySQL connections. Check your log file for specific errors
```

I uncommented Logger.sql.driver and then ran it all again.

```
Opening DatabasePool 'acore_auth'. Asynchronous connections: 1, synchronous connections: 1.
Could not connect to MySQL database at 192.168.0.118: Host 'host.docker.internal' is not allowed to connect to this MySQL server
```

I tried following the clues to an instruction to use named pipes on Windows and "enable-named-pipe" to [mysqld] in section my.ini, by unhashing enable-named-pippe under the server section but that didn't work either.

I tried installing MySQL from Windows but now the server opens and shuts too darn fast for me to even see the errors, so it seems to have gotten worse, not better.

**Comment by Kuiriel** [comment]
I figured I broke something, so I reset the conf files entirely. It looks like the issue is I am an idiot and broke something somewhere by trying to set IP addresses up, and it's all me and not you.
